### PR TITLE
Strip rotated client secret from export json

### LINKED ISSUE #19373: Realm export contains rotated client secret
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the latest release

### Area

import-export

### Describe the bug

When creating a partial export of a realm the resulting json contains the "client.secret.rotated" attribute in "attributes" for an exported client (of course only when the feature 'client-secret-rotation' is enabled). This secret is not redacted/stripped.

### Version

20.0.3

### Expected behavior

I expected a redacted rotated client secret (`**********`), just like with the (non-rotated) client secret.

### Actual behavior

The unredacted rotated client secret is in the exported json

### How to Reproduce?

- start keycloak with `--features=client-secret-rotation`
- create a client profile with "executor": "secret-rotation"
- create a client policy
- client client and generate new secret
- partial export of realm setting, check the 'clients' checkbox

### Anything else?

I'll create a PR with the fix

### PR DESCRIPTION
**Pull request by rwiskerke**
Closes #19373

### CODE REVIEW

**Review by martin-kanis — CHANGES_REQUESTED** [review]
Thank you for the PR @rwiskerke. Looks good, but could you also test this in integration tests? The easiest way would be to add 
```
// client rotated secret in attributes
if (client.getAttributes() != null && client.getAttributes().containsKey(ClientSecretConstants.CLIENT_ROTATED_SECRET)) {
    Assert.assertEquals("Client rotated secret masked", ComponentRepresentation.SECRET_VALUE, client.getAttributes().get(ClientSecretConstants.CLIENT_ROTATED_SECRET));
}
```
into `PartialExportTest#checkSecretsAreMasked` and 

```
"client.secret.rotated.creation.time": "1680684475",
"client.secret.rotated.expiration.time": "1680857275",
"client.secret.rotated": "zxJA2NT3Qoic1Z7SjKK2l06sLQLpLM56"
```
into `testsuite/integration-arquillian/tests/base/src/test/resources/export/partialexport-testrealm.json` to i.e `test-app-service-account` client.

**Comment by rwiskerke** [comment]
> @rwiskerke Will you be able to update the PR anytime soon?

I'm afraid not, maybe next week I'll have some more time.

**Comment by rwiskerke** [comment]
@martin-kanis I added the integration test and rebased on current main

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/partialexport/PartialExportTest.java [hunk 1]
```diff
@@ -15,11 +16,8 @@
 import org.keycloak.testsuite.ProfileAssume;
 import org.keycloak.testsuite.admin.AbstractAdminTest;
 
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
```
Comment (martin-kanis) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/partialexport/PartialExportTest.java:19]: Can you please remove the star import? We don't use them in Keycloak.

**Review by martin-kanis — CHANGES_REQUESTED** [review]
@rwiskerke Thank you for adding the integration test. Just one minor thing:

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/partialexport/PartialExportTest.java (hunk 1 above)
Comment (rwiskerke) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/partialexport/PartialExportTest.java]: sure, I guess my IDE did that when I used Optional

Comment (martin-kanis) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/partialexport/PartialExportTest.java]: Can you also squash the commits please? Then it's ready from my POV.

Comment (rwiskerke) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/partialexport/PartialExportTest.java]: Done!

**Review by martin-kanis — APPROVED** [review]
@rwiskerke Thank you! LGTM

**Review by mhajas — APPROVED** [review]
Thanks @rwiskerke, LGTM

**Pull request merged**